test/spatial_test_networks.jl

```diff
@@ -0,0 +1,186 @@
+### Fetch packages ###
+using Catalyst, Graphs
+
+### Helper Functions ###
+
+# Generates ranomised intiial condition or paraemter values.
+rand_v_vals(grid) = rand(nv(grid))
+rand_v_vals(grid, x::Number) = rand_v_vals(grid) * x
+rand_e_vals(grid) = rand(ne(grid))
+rand_e_vals(grid, x::Number) = rand_e_vals(grid) * x
+function make_u0_matrix(value_map, vals, symbols)
```

Comment:
A good practice is to keep randomness in tests as minimal as possible, because randomness leads to flaky tests. Sometimes it is unavoidable because the thing being tested is random but we should not introduce new randomness. Randomness also makes the test easier to understand, which is a problem, because tests are themselves untested. 
If you feel that you have to generate the initial conditions randomly, set a seed to make the test reproducible.

Reply:
Okay, having a seed is good. I would still recommend using hard-coded values though. If you truly don't care which choice the rng makes, just make the choice yourself. Or even just hard-code the values that are produced by this seed. There are several benefits to this:
1. Test breakage. RNGs sometimes change implementation under the hood, which leads to the same seed producing different randomness. This might break the test or make people afraid it will break the test.
2. Test readability. Random initialization unnecessarily introduces additional cognitive load on the reader of the code. Code is read **at least** 10 times more often than it is written. We should not make the reader dig in and run this test themselves just to learn which values this seed generates.

Reply:
StableRngs are explicitly for uses like CI, and are promised to remain stable for a given seed across releases. So they will return the same sequence of random numbers every time. That’s why we use it for code that needs a `rng`. That said, to use the stable rng one has to call  always call`rand(rng)`, otherwise the Julia default rng is called, so the stable rng isn’t being used here. (Independently, as you say it isn’t clear if one really needs to use a random value in this case either.)

Reply:
`rng` is nowed used properly.